### DISCUSSION TITLE
EUM Server - Questions

### DISCUSSION THREAD

**Discussion by nevesing** [inspectIT/inspectit-ocelot#1218] [Q&A]
Hello,

I am trying to use the EUM component only and I am able to successfully see the beacon metrics in prometheus endpoint(8888). My questions are:

Startup command:

`java -Dserver.port=7000 -jar=/Data/poc/inspectit-ocelot-eum-server-1.3.jar`

Beacon URL: https://myhost:7000/beacon

1. Can we change the prometheus port no from 8888 to something else (I have firewall issue)
2. I am using my customized version of boomerang.js; so how do I use that to overwrite the version provided by EUM server
3. Is there a way to write the beacon data to the filesystem as JSON

**Comment by mariusoe** [comment]
Hi @nevesing 

> Can we change the prometheus port no from 8888 to something else (I have firewall issue)

yes, you can easily change the Prometheus port by setting the `inspectit-eum-server.exporters.metrics.prometheus.enabled.host.port`. See also [the exporter documentation](https://inspectit.github.io/inspectit-ocelot/docs/enduser-monitoring/eum-server-configuration#exporters).

>I am using my customized version of boomerang.js; so how do I use that to overwrite the version provided by EUM server

The Boomerang which is provided by the EUM server is located in its JAR file and there is currently no way to replace it (only you modify the JAR file). Anyway, we just included Boomerang for convenience, so you don't have to host one by yourself. If you have a custom one, you have to host it by yourself.

> Is there a way to write the beacon data to the filesystem as JSON

No. Unfortunately, there isn't currently an out-of-the-box option for this.
But it would be quite easy to implement this by yourself. For example, you could extend the [rocks.inspectit.oce.eum.server.rest.BeaconController#processBeacon](https://github.com/inspectIT/inspectit-ocelot/blob/master/components/inspectit-ocelot-eum-server/src/main/java/rocks/inspectit/oce/eum/server/rest/BeaconController.java) to serialize each received Beacon as a JSON object and export it to disk.

**Comment by nevesing** [comment]
I had to slightly modify the java argument:

`java -Dserver.port=7000 -Dinspectit-eum-server.exporters.metrics.prometheus.port=8088 -Dinspectit-eum-server.exporters.metrics.prometheus.host=my_host_name -jar /Data/poc/inspectIT/inspectit-ocelot-eum-server-1.3.jar`

Regarding hosting boomerang.js via inspectIT, can I manually build the EUM source by changing the boomerang.js file? Just want to know the feasibility.

**Comment by nevesing** [comment]
Thanks for making this project and it is very easy to implement the EUM server. My current challenge is  that I wanted to implement boomerang.js in multiple applications (say 50) and collect metrics from 5 different plugins. If all the metrics from 50 applications are exposed to same prometheus exporter (port 8088), I am not sure how to play around with the output. Also I need to update the config YAML for every metric. It is good to have simple JSON exporter to write to disk.
